models/discordactions.js

```diff
@@ -87,9 +90,34 @@ const addGroupRoleToMember = async (roleData) => {
   }
 };
 
+/**
+ *
+ * @param userDiscordId { String }: DiscordId of the user
+ * @returns {Promise<String>}
+ */
+const updateDiscordImageForVerification = async (userDiscordId) => {
+  try {
+    const discordAvatarUrl = await generateDiscordProfileImageUrl(userDiscordId);
+    const verificationDataSnapshot = await photoVerificationModel.where("discordId", "==", userDiscordId).get();
+    const unverifiedUserDiscordImage = {
+      discord: { url: discordAvatarUrl, approved: false, date: admin.firestore.Timestamp.fromDate(new Date()) },
+    };
+    if (verificationDataSnapshot.empty) {
+      throw new Error("No user verification record found");
+    }
+    const documentRef = verificationDataSnapshot.docs[0].ref;
```

Comment:
since we are letting out single data do we need the DTO from verificationDataSnapshot to be an array?

Reply:
Thats how firebase works and has always worked